src/numerics/Func1.cpp

```diff
@@ -223,6 +223,39 @@ Poly1::Poly1(const vector<double>& params)
     copy(params.data(), params.data() + m_cpoly.size(), m_cpoly.begin());
 }
 
+string Poly1::write(const string& arg) const
+{
+    // write terms in reverse order
+    string out = "";
+    if (m_cpoly[m_cpoly.size()-1] != 0.) {
+        out = fmt::format("{}", m_cpoly[m_cpoly.size()-1]);
+    }
+    for (size_t n=1; n<m_cpoly.size(); n++) {
+        if (m_cpoly[m_cpoly.size()-1-n] == 0.) {
+            continue;
+        }
+        string term;
+        if (m_cpoly[m_cpoly.size()-1-n] == 1.) {
+            term = fmt::format("{}", arg);
+        } else if (m_cpoly[m_cpoly.size()-1-n] == -1.) {
+            term = fmt::format("-{}", arg);
+        } else {
+            term = fmt::format("{}{}", m_cpoly[m_cpoly.size()-1-n], arg);
+        }
+        if (n > 1) {
+            term = fmt::format("{}^{}", term, n);
```

Comment:
If this is meant to be a LaTeX string, then the exponent should be surrounded with (escaped) braces, in case you have a polynomial of order 10 or more.